File: lib/Config/INIParser.cpp
#include <fstream>
#include <string>
#include "ConnectionSettings.hpp"
#include "UnBoard.hpp"

void loadConfig() {
    std::ifstream file("config.ini");
    if (!file)
        throw std::runtime_error("Fail to open config.ini");

    std::string line;
    while (std::getline(file, line)) {
        auto pos = line.find('=');
        if (pos == std::string::npos) continue;

        std::string key = line.substr(0, pos);
        std::string value = line.substr(pos + 1);

        if (key == "NAOqiIP") naoqiAddress.ip = value;
        else if (key == "NAOqiPort") naoqiAddress.port = std::stoi(value);
        else if (key == "gameControllerIP") gameControllerAddress.host = value;
        else if (key == "writingPort") gameControllerAddress.writingPort = std::stoi(value);
        else if (key == "readingPort") gameControllerAddress.readingPort = std::stoi(value);
        else if (key == "broadcast") teamCommunicationAddress.broadcast = value;
        else if (key == "multicast") teamCommunicationAddress.multicast = value;
        else if (key == "teamPort") teamCommunicationAddress.teamPort = std::stoi(value);
        else if (key == "unbeatablesNumber") unbeatablesReturnBoard.teamNum = std::stoi(value);
        else if (key == "playerNumber") {
            unbeatablesReturnBoard.playerNum  = std::stoi(value);
            selfMessage.playerNumber = std::stoi(value);
        }
        else if (key == "camera") systemSettings.camera = std::stoi(value);
        else if (key == "cascade") systemSettings.cascade = value;
        else if (key == "controlDevice") controlState.device = value;
    }
}

File: Libs/Ini/iniParser.py
import UnBoard

class ConfigLoader(object):
    def __init__(self, filename):
        self.filename = filename

    def apply(self):
        try:
            f = open(self.filename, "r")
        except IOError:
            raise RuntimeError("Fail to open {}".format(self.filename))

        for line in f:
            line = line.strip()
            if not line or line.startswith("#") or line.startswith("["):
                continue

            if '=' not in line:
                continue

            key, value = line.split('=', 1)
            key = key.strip()
            value = value.strip()

            if key == "NAOqiIP":
                UnBoard.naoqiAddress_ip = value
            elif key == "NAOqiPort":
                UnBoard.naoqiAddress_port = int(value)
            elif key == "gameControllerIP":
                UnBoard.gameControllerAddress_host = value
            elif key == "readingPort":
                UnBoard.gameControllerAddress_readingPort = int(value)
            elif key == "writingPort":
                UnBoard.gameControllerAddress_writingPort = int(value)
            elif key == "broadcast":
                UnBoard.teamCommunicationAddress_broadcast = value
            elif key == "multicast":
                UnBoard.teamCommunicationAddress_multicast = value
            elif key == "teamPort":
                UnBoard.teamCommunicationAddress_teamPort = int(value)
            elif key == "unbeatablesNumber":
                UnBoard.unbeatablesReturnBoard_teamNum = int(value)
            elif key == "playerNumber":
                UnBoard.unbeatablesReturnBoard_playerNum = int(value)
                UnBoard.selfMessage_playerNumber = int(value)
            elif key == "camera":
                UnBoard.systemSettings_camera = int(value)
            elif key == "cascade":
                UnBoard.systemSettings_cascade = value
        f.close()
